File: LeetCode/3379-score-of-a-string/solution.java
class Solution {
    public int scoreOfString(String s) {
       return calc(s,0);
    }
    private int calc(String s, int idx){
        if(idx == s.length()-1){
            return 0;
        }
        int diff = Math.abs(s.charAt(idx) - s.charAt(idx+1));
        return diff + calc(s,idx+1);
    }
}

File: LeetCode/3379-score-of-a-string/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var scoreOfString = function(s) {
        return calc(s,0);
    }
   function calc(s,i){
        if(i === s.length-1){
            return 0;
        }
        let diff = Math.abs(s.charCodeAt(i) - s.charCodeAt(i+1));
        return diff + calc(s,i+1);
    }

